#! /usr/bin/env python

PACKAGE='ece6460_lane_detection_example'

from dynamic_reconfigure.msg import SensorLevels
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       Name                Type      Level     Description                             Default   Min     Max
gen.add("h_center",         int_t,    0,        "Hue threshold center",                 27,       0,      180)
gen.add("h_width",          int_t,    0,        "Hue threshold width",                  6,        1,      90)
gen.add("val_thres",        int_t,    0,        "Value threshold",                      185,      0,      255)
gen.add("sat_thres",        int_t,    0,        "Saturation threshold",                 70,       0,      255)
gen.add("mask_height",      int_t,    0,        "Mask height",                          296,      0,      767)
gen.add("cluster_tol",      double_t, 0,        "Distance to include point in cluster", 3.0,      0.1,    10.0)
gen.add("min_cluster_size", int_t,    0,        "Minimum points in a cluster",          20,       1,      1000)
gen.add("max_cluster_size", int_t,    0,        "Maximum points in a cluster",          500,      100,    2000)
gen.add("fit_order",        int_t,    0,        "Polynomial order for curve fitting",   2,        1,      6)
gen.add("rms_tolerance",    double_t, 0,        "RMS error threshold",                  0.2,      0.001,  10.0)

exit(gen.generate(PACKAGE, PACKAGE, "LaneDetection"))
